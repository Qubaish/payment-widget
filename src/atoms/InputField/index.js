import React, { useState, memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Input = styled.input`
    width: 25px;
    background: #D1CEC7;
    border: none;
    border-bottom: 1px solid #B9B8B7;
    font-size: 1.6rem;
    text-align: center;
    :focus {
        outline: none;
        border-bottom: 1px solid #307EF7;
    }
`;

function InputField({placeholder, onChange, index}){
    const [state, setState] = useState('');

    const validateInput = val => {
        const number = isNaN(val) ? '' : val;
        return number && number.length > 1 ? number.substr(0, 1) : number;
    }

    const onHandleChange = e => {
        const validateNumber = validateInput(e.target.value);
        setState(validateNumber);
        onChange(validateNumber, index);
    }

    return (
        <Input
            placeholder={placeholder}
            data-testid="input"
            onChange={onHandleChange}
            value={state}
            aria-label="Enter OTP digit"
        />
    );
}

InputField.propTypes = {
    placeholder: PropTypes.string,
};

InputField.defaultProps = {
    placeholder: '0',
}

export default memo(InputField);