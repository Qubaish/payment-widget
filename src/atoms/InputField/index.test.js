import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import InputField from './index';

describe('<InputField />', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<InputField placeholder="0" onChange={jest.fn()}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render a InputField', () => {
        const { getByTestId} = render(<InputField placeholder="0" onChange={jest.fn()}/>);
        const elem = getByTestId('input');
        expect(elem.placeholder).toBe('0');
    });
    it('should check onChange functionality', () => {
        const { getByTestId} = render(<InputField placeholder="0" onChange={jest.fn()}/>);
        const elem = getByTestId('input');
        fireEvent.change(elem, {target: {value: '5'}});
        expect(elem.value).toBe('5');
    });
    it('should check input should be integer', () => {
        const { getByTestId} = render(<InputField placeholder="0" onChange={jest.fn()}/>);
        const elem = getByTestId('input');
        fireEvent.change(elem, {target: {value: 'S'}});
        expect(elem.value).toBe('');
    });
});
