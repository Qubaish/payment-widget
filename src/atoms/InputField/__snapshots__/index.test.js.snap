// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> checking snapshot 1`] = `
<DocumentFragment>
  <input
    aria-label="Enter OTP digit"
    class="sc-AxjAm bPsFAW"
    data-testid="input"
    placeholder="0"
    value=""
  />
</DocumentFragment>
`;
